WifiART/src/structs/ArtSystemState.java

```diff
@@ -121,16 +120,94 @@ public LcrState getLcrState() {
 
     /**
      * Gets the subelement state for the Map Image subelement.
+     *
      * @return the Map Image subelement state
      */
     public MapState getMapState() {
         return mapState;
     }
 
+
+    // Getters for which subelements are included
+
+    /**
+     * Gets whether or not a given subelement is to be included in the output.
+     *
+     * @param subelementName the subelement being inquired about
+     * @return whether or not the given subelement should be included in the output.
+     */
+    public boolean isSubelementIncluded(SubelementName subelementName) {
+        return includedSubelements.get(subelementName);
```

Comment:
Is it guaranteed includedSubelements will contain all possible subelement names when this method is called? If not then this could throw NPE if the subelementName doesn't exist in includedSubelements.

Reply:
Yes, it is guaranteed. All possible subelement names are initialized in the init() method.